File: 프로그래머스/lv2/42584. 주식가격/주식가격.py
from collections import deque

def solution(prices):
    l = len(prices)
    answer = [0] * l
    prices = deque(prices)

    for i in range(l):
        now_price = prices.popleft()
        for price in prices:
            answer[i] += 1
            if price < now_price:
                break

    return answer

File: 프로그래머스/lv2/42584. 주식가격/주식가격.java
class Solution {
    public int[] solution(int[] prices) {
        int n = prices.length;
        int[] answer = new int[n];

        for (int i = 0; i < n - 1; i++) {
            int currentPrice = prices[i];
            int time = 0;

            for (int j = i + 1; j < n; j++) {
                if (prices[j] >= currentPrice) {
                    time++;
                } else {
                    time++;
                    break;
                }
            }

            answer[i] = time;
        }

        answer[n - 1] = 0;  

        return answer;
    }
}
